_P172_PZEM-004Tv3.ino

```diff
@@ -0,0 +1,873 @@
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+        uint8_t slave_addr = (uint8_t)PCONFIG(0);
+        
+        if((P172_first_init_done) && (event->TaskIndex != P172_first_unit_index))
+        {
+            addHtml(F("<br><span style=\"color:darkgreen;\">Using serialport settings from first PZEM004Tv3 device</span>"));
+        }
+        else
+        {
+            addHtml(F("<TR><TD>ESP RX pin:<TD>"));
+            addPinSelect(false, F("taskdevicepin1"), CONFIG_PIN1);
+            addHtml(F("<TR><TD>ESP TX pin:<TD>"));
+            addPinSelect(false, F("taskdevicepin2"), CONFIG_PIN2);
+            
+            //addFormNote(F("SoftSerial: 1st=RX-Pin 14 2nd=TX-Pin 12"));
+        }
+        
+        addFormNumericBox(F("Device Address"), F("p172_slave_addr"), slave_addr, 0, 248);
+        
+        addHtml(F("</tr><tr><td></td><td \"height:20px;\"><p style=\"font-size:10pt\">Enter device address in decimal (0~248). <span style=\"color:#ff0000;\">Use 0 only when using just one device.</span></p></td></tr>"));
+        
+        addHtml(F("<tr><td></td><td \"height:20px;\"><p style=\"font-size:10pt;\">To use multiple devices on one UART port, wire them like <a style=\"font-size:10pt;\" href=\"https://github.com/Jansemar/PZEM004Tv30MULTI/raw/master/multiuartdevs.png\" target=\"_blank\" rel=\"noopener\">this</a></p></td></tr>"));
+        addHtml(F("<tr><td></td><td \"height:20px;\"><p style=\"font-size:10pt;\"><span style=\"color:#ff0000;\">WARNING </span>The PZEM outputs 5V and the ESP can handle only 3.3V Use a levelshifter, </p></td></tr>"));
+        addHtml(F("<tr><td></td><td \"height:20px;\"><p style=\"font-size:10pt;\">voltage divider or mod your PZEM to prevent damage to your ESP.</p></td></tr>"));
+        addFormSubHeader(F("How to assign an address and clear energy value"));
+        addHtml(F("<tr><td colspan=\"2; height:20px;\"><p style=\"font-size:10pt;\">To set a device address, connect the device you want to use to your ESP using the pins you configured at the first pzem device.</p></td></tr>"));
+        addHtml(F("<tr><td colspan=\"2; height:20px;\"><p style=\"font-size:10pt;\">Go to the TOOLS tab on your ESPEasy and enter the command: <strong><em>pzemsetaddr,X</em></strong> where X is the address u want to set.</p></td></tr>"));
+        addHtml(F("<tr><td colspan=\"2; height:20px;\"><p style=\"font-size:10pt;\">To reset the total power value enter the command: <strong><em>resetpzemenergy,X</em></strong> where X is the device address u want to reset.</p></td></tr>"));
+        addHtml(F("<tr><td colspan=\"2; height:20px;\"><p style=\"font-size:10pt;\">If you enter <strong><em>resetpzemenergy</em></strong> without an address the power values from all devices will be set to 0.</p></td></tr>"));
+        addHtml(F("<tr><td colspan=\"2; height:20px;\"><p style=\"font-size:10pt;\">All commands can also be executed the usual ESPEasy way trough web or MQTT. See ESPEasy docs for more information.</p></td></tr>"));

```

Comment:
This is a lot of HTML code.
If you need special explanation, then the docs are the place for it, not all in a plugin.

FYI: we plan to move (quite soon) to a new UI, where the UI does collect data from a plugin via JSON. So lots of extra HTML not only add up to the size of the binary, but also are quite specific for the current CSS style etc.